catroid/src/main/java/org/catrobat/catroid/ui/adapter/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui.adapter;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageSwitcher;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.ViewSwitcher;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.copypaste.ListItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecyclerViewAdapter<T> extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {
+
+	public static final String TAG = RecyclerViewAdapter.class.getSimpleName();
+
+	private static final int SELECTED_BG_COLOR = 0xFFDDDDDD;
```

Comment:
this has to be changed once the UX team's mind is made up about the colours (so this is basically just a temporary way to set it). additionally there is probably more the UX team wants to change for the selected items, so I thought for the time being, this should suffice.